webserver/src/main/java/io/kestra/webserver/controllers/NamespaceFileController.java

```diff
@@ -155,6 +158,17 @@ public void delete(
         storageInterface.delete(tenantService.resolveTenant(), toNamespacedStorageUri(namespace, path));
     }
 
+    @ExecuteOn(TaskExecutors.IO)
+    @Get(uri = "files/distinct-namespaces", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Files"}, summary = "List existing namespace folders")
+    public List<String> distinctNamespaces() throws IOException, URISyntaxException {
+        return storageInterface.list(tenantService.resolveTenant(), null)
+            .stream()
```

Comment:
Already said (and explained why) it's a bad idea but I will not block the PR for that if nobody else object.